flash/data/process.py

```diff
@@ -101,6 +101,151 @@ class PreprocessState:
 
 
 class Preprocess(Properties, torch.nn.Module):
+    """
+    The :class:`~flash.data.process.Preprocess` is used to encapsulate
+    all the processing data loading logic up to the model.
+
+    It is particularly relevant when you want to provide an end to end implementation which works
+    with 4 different stages: ``train``, ``validation``, ``test``,  and inference (``predict``).
+
+    The :class:`~flash.data.process.Preprocess` supports the following hooks:
```

Comment:
Let's just add here something like:
You can override any of the preprocessing hooks to provide custom functionality. All hooks default to no-op (except....)

And then add a snippet example of a full custom Preprocess.

All the hooks can be explained in their own autodocs.

Reply:
Let's clean this out in another PR.